hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/metrics/DataNodeDiskMetrics.java

```diff
@@ -127,6 +143,16 @@ public void run() {
 
             detectAndUpdateDiskOutliers(metadataOpStats, readIoStats,
                 writeIoStats);
+
+            // Sort the slow disks by latency.
+            if (maxSlowDisksToBeExcluded > 0) {
+              ArrayList<DiskLatency> diskLatencies = new ArrayList<>();
+              for (Map.Entry<String, Map<DiskOp, Double>> diskStats :
+                  diskOutliersStats.entrySet()) {
+                diskLatencies.add(new DiskLatency(diskStats.getKey(), diskStats.getValue()));
+              }
+              sortSlowDisks(diskLatencies);
```

Comment:
`sortSlowDisks` seems to do more than just sort slow disks, it also limits them with `maxSlowDisksToBeExcluded` and sets them to `slowDisksToBeExcluded`. It would be better to give this method a more appropriate name.

Reply:
You could use `Collections.sort` to make it more concise.
```suggestion
              Collections.sort(diskLatencies, (o1, o2)
                  -> Double.compare(o2.getMaxLatency(), o1.getMaxLatency()));
              slowDisksToBeExcluded = diskLatencies.stream().limit(maxSlowDisksToBeExcluded)
                  .map(DiskLatency::getSlowDisk).collect(Collectors.toList());
```

Reply:
Thank you @tasanuma very much for your review and comments. I think your suggestions are very good. I will revise ASAP.